File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.java
class RandomizedSet {


    ArrayList<Integer> nums; 
    HashMap<Integer , Integer > map ; 
    java.util.Random rand = new java.util.Random();

    public RandomizedSet() {
        nums = new ArrayList<Integer>();
      map = new HashMap<Integer , Integer>();
    }
    
    public boolean insert(int val) {
         boolean bool = map.containsKey(val);
      if( bool) return false;
      map.put(val , nums.size()) ;
      nums.add(val) ;
      return true ;
    }
    
    public boolean remove(int val) {
            boolean bool = map.containsKey(val); 
    if( !bool) return false ; 
    int m = map.get(val) ; 
    if( m < nums.size() -1 ) { 
        int last = nums.get(nums.size() -1) ; 
        nums.set(m , last) ; 
        map.put( last , m) ; 
    }
    map.remove(val); 
    nums.remove(nums.size() -1) ; 
    return true ; 
    }
    
    public int getRandom() {
        return nums.get( rand.nextInt(nums.size()) ) ;
    }
}

/**
 * Your RandomizedSet object will be instantiated and called as such:
 * RandomizedSet obj = new RandomizedSet();
 * boolean param_1 = obj.insert(val);
 * boolean param_2 = obj.remove(val);
 * int param_3 = obj.getRandom();
 */

File: 0380-insert-delete-getrandom-o1/0380-insert-delete-getrandom-o1.py
class RandomizedSet:

    def __init__(self):
        self.box = set() 

    def insert(self, val: int) -> bool:
        if val not in self.box: 
            self.box.add(val)
            return True 
        else: 
            return False 

    def remove(self, val: int) -> bool:
        if val in self.box: 
            self.box.discard(val)
            return True 
        else: 
            return False 
        

    def getRandom(self) -> int:
        return choice( list(self.box))
        


# Your RandomizedSet object will be instantiated and called as such:
# obj = RandomizedSet()
# param_1 = obj.insert(val)
# param_2 = obj.remove(val)
# param_3 = obj.getRandom()